pkg/controller/controller.go

```diff
@@ -742,6 +758,21 @@ func (p *csiProvisioner) prepareProvision(ctx context.Context, claim *v1.Persist
 		deletionAnnSecrets.namespace = provisionerSecretRef.Namespace
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.VolumeAttributesClass) && vacName != nil {
+		vac, err := p.client.StorageV1alpha1().VolumeAttributesClasses().Get(ctx, *vacName, metav1.GetOptions{})
+		if err != nil {
+			return nil, controller.ProvisioningNoChange, err
+		}
+
+		if vac.DriverName != p.driverName {
+			mismatchErrorMsg := fmt.Sprintf("VAC %s referenced in PVC is for driver %s which does not match driver name %s", *vacName, vac.DriverName, p.driverName)
+			p.eventRecorder.Event(claim, v1.EventTypeWarning, "Provisioning", mismatchErrorMsg)
```

Comment:
sig-storage-lib-external-provisioner [will already send an event](https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/blob/cd540b4cb6c7b01eaad8434c273d00aa13ce9081/controller/controller.go#L1474) when `Provision()` returns an error, no need to send another one here.